### PR TITLE
completes getAllTopics request

### PR DESCRIPTION
**Pull request by patrickarigg**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const express = require("express");
+const { getAllTopics } = require("./controllers/app.controller");
+const { handlePostgresErrors, handleCustomErrors, handleServerErrors } = require("./errors");
+
+const app = express();
+app.use(express.json());
```
Comment (johnhutcheon) [inline app.js]: We're not currently using this line of code, so it can be removed for now 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+describe("/api/topics", () => {
+  test("GET 200: should return array of all topics with required properties", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const topics = body.topics;
+        expect(Array.isArray(topics)).toBe(true);
+        expect(topics).toHaveLength(3)
```
Comment (johnhutcheon) [inline __tests__/app.test.js:18]: Nice work checking the length before a forEach 🙌

File: controllers/app.controller.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const { selectAllTopics } = require("../models/app.model");
+
+exports.getAllTopics = (req, res, next) => {
+  selectAllTopics()
+    .then(({ rows }) => {
+      const topics = rows;
```
Comment (johnhutcheon) [inline controllers/app.controller.js]: Anything database related should really be handled in the model. If you return the rows from your model, you should only need to call selectAllTopics here, then send the response. 👍

File: errors.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+exports.handlePostgresErrors = (err, req, res, next) => {
+  next(err)
+};
+
+exports.handleCustomErrors = (err, req, res, next) => {
+  next(err);
+};
```
Comment (johnhutcheon) [inline errors.js]: Good that you're thinking ahead, but as these error handlers are currently not doing anything, can they be removed until later please? 😄

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Patrick, nice start! Could you just see my comments please and make the amendments I've requested then resubmit? Thanks! 👍

File: app.js (hunk 1 above)
Comment (patrickarigg) [reply app.js]: removed in latest commit

File: controllers/app.controller.js (hunk 3 above)
Comment (patrickarigg) [reply controllers/app.controller.js]: updated

File: errors.js (hunk 4 above)
Comment (patrickarigg) [reply errors.js]: removed postgres and custom handler

**Pull request merged**